hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1780,6 +1781,29 @@ public OptionalDouble getAvgStoreFileAge() {
     return getStoreFileAgeStream().average();
   }
 
+  @Override
+  public Map<String, Pair<Long, Long>> getStoreFilesAccessTimeAndSize() {
+    Collection<HStoreFile> storeFiles = this.storeEngine.getStoreFileManager().getStorefiles();
+    Map<String, Pair<Long, Long>> sfAccessTimeAndSizeMap = new HashMap<>();
+    for (HStoreFile sf : storeFiles) {
+      if (sf.getReader() == null) {
+        continue;
+      }
+      FileStatus fileStatus;
+      try {
+        fileStatus = sf.getFileInfo().getFileStatus();
```

Comment:
If this FileStatus is cached, then the accessTime will not be updated?

Reply:
> If this FileStatus is cached, then the accessTime will not be updated?

@Apache9 sir. Thanks for your review. 
Yes, if FileStatus is cached, the accessTime will not be updated, but the current logic is got from the back-end file system in real time.

Reply:
Will this introduce a request spike to NN? Usually after opening a store file we will not goto NN any more...

I wonder if we can maintain this information inside HBase?